ffi/cpp/include/utils.hpp

```diff
@@ -0,0 +1,43 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#ifndef ICU4X_UTILS_HPP
+#define ICU4X_UTILS_HPP
+
+#include<string>
+#include "../../capi/include/custom_writeable.h"
+
+namespace icu4x {
+namespace internal {
+
+extern "C" inline void Flush(ICU4XWriteable* w) {
+    std::string* string = reinterpret_cast<std::string*>(w->context);
+    string->resize(w->len);
+};
+
+extern "C" inline bool Grow(ICU4XWriteable* w, uintptr_t requested) {
+    std::string* string = reinterpret_cast<std::string*>(w->context);
+    string->resize(requested);
+    w->cap = string->length();
+    w->buf = string->data();
+    return true;
+};
+
+inline ICU4XWriteable WriteTo(std::string& string) {
```

Comment:
suggestion: Maybe naming this something like `AsWriteable` or `WriteableFromString` might capture the intention a bit better here.